### PR TITLE
Refactor GitHub actions + fix build

### PR DESCRIPTION
**Pull request by jemand771**
This PR supersedes #17. I haven't worked on that branch in ages and starting from scratch was just easier

this still needs some cleanup work, e.g. to go.mod/go.sum.
I also haven't decided how much time I want to spend rewording and squashing commits yet. I know "I hate bash" isn't very descriptive, but at this point I just want to get this out
_also_ also: docs - I've already started writing some maintainer docs, but I don't want to block this with things that can be done later. we'll see

Since reviewing this is kinda impossible, I'll just merge (and test here) once I feel like it's ready. Test runs over on my fork are looking good so far. Also, finding issues during real use is way easier than using artificial testing scenarios. If anyone has any objections or comments, now's the time.

I've said this in a couple of places already, but sorry to everyone this took so long. It was a lot of work, but not nearly enough to justify taking over a year to do it. either way, here we are. Statping-ng soon™

### CODE REVIEW

**Comment by dhkatz** [comment]
Thank you for the work on this. Much appreciated! 💖

**Comment by jemand771** [comment]
ah, screw it. this is never gonna be perfect and at this point I just want to get some things done

I'll try to make a test release with nothing really new in it, just to see if everything also works _here_
after that, things that are still due:
* dockerhub/quay.io/... docker image?
* multiplatform docker image (currently only `linux/amd64`)
* go crossbuild for all platforms (iirc some were broken and/or disabled)
* test job (has a bunch of external dependencies)
* incremental change list?
  * rc to rc doesn't list all changes again
  * it should (hopefully) do that for rc->stable, based on the previous stable release

**Comment by jemand771** [comment]
I just played around with xgo a bit and here's what I found out about broken builds:
* `darwin/386` was [dropped with go 1.15](https://go.dev/doc/go1.15#darwin) - xgo reports a rather ominous "go version too high" if you try it anyway
* the same announcement also hints that `amd64` and `arm64` are the only supported targets for darwin. since those already work, darwin builds already work as well as they possibly could
* all targets work on linux (😎)
* windows: [crazy-max' goxx](https://github.com/crazy-max/goxx#supported-platforms) only supports `windows/386` and `windows/amd64`, which both work here already

tl;dr: all (common) officially supported xgo build targets already work, just the docker image is missing multi-archification
